corelib-utils/src/test/java/eu/europeana/corelib/utils/DateUtilsTest.java

```diff
@@ -32,4 +32,40 @@ public void formatTest(){
 	public void parseTest(){
 		assertEquals(10000000L,DateUtils.parse("1970-01-01T02:46:40.000Z").getTime());
 	}
+
+	@Test
+	public void test_isYearOrYearRange() {
+		//check all possibilities for year range value
+		assertTrue(DateUtils.isYearRange("1880-1990"));
+		assertFalse(DateUtils.isYearRange("2000-1990"));
+		assertFalse(DateUtils.isYearRange("Geschichte 1775-1798"));
+		assertFalse(DateUtils.isYearRange("Geschichte 1775"));
+		assertFalse(DateUtils.isYearRange("1775 -Geschichte"));
+		assertFalse(DateUtils.isYearRange("1775 -"));
+
+		//check year
+		assertTrue(DateUtils.isYear("1880"));
+		assertFalse(DateUtils.isYear("188F"));
+		assertFalse(DateUtils.isYear("1C80"));
+		assertFalse(DateUtils.isYear("abcd"));
+	}
+
+	@Test
+	public void test_isIsoDate() {
+		//yyyy-MM-dd format check
+		assertTrue(DateUtils.isIsoDate("2000-02-13"));
+		assertFalse(DateUtils.isIsoDate("13-02-2000"));
+		assertFalse(DateUtils.isIsoDate("testing-90"));
+		assertFalse(DateUtils.isIsoDate("8765 testing"));
+	}
+
+	@Test
+	public void test_isIsoDateTime () {
+		//yyyy-MM-dd'T'HH:mm:ss.SSS'Z'  Date time check
+		assertTrue(DateUtils.isIsoDateTime("2017-07-26T01:00:00.000Z"));
+		assertFalse(DateUtils.isIsoDateTime("2017-07-26T01:00:00Z"));
+		assertFalse(DateUtils.isIsoDateTime("2017-07-2601:00:00.000Z"));
+		assertFalse(DateUtils.isIsoDateTime("2017-07-26T01:00:00.000"));
+		assertFalse(DateUtils.isIsoDateTime("2017-07-26T01::00.000Z"));
```

Comment:
It's good that you're testing a positive and negative example, but it's always a good idea to also check what happens if you provide null or an empty string, especially because Europeana record data is not consistent and in some collections values are missing

Reply:
Have added the test for null and empty string value. Also have added the null checks in DateUtils